django/db/backends/postgresql/base.py

```diff
@@ -179,6 +197,45 @@ class DatabaseWrapper(BaseDatabaseWrapper):
     ops_class = DatabaseOperations
     # PostgreSQL backend-specific attributes.
     _named_cursor_idx = 0
+    _connection_pools = {}
+
+    @property
+    def pool(self):
+        if not is_psycopg3:
+            return None
+
+        if self.alias not in self._connection_pools:
+            pool = None
+            pool_options = self.settings_dict.get("OPTIONS", {}).get("pool", None)
+
+            if pool_options is not None:
+                # Verify that we are not running with persistent connections
+                if self.settings_dict.get("CONN_MAX_AGE", 0) != 0:
+                    raise ImproperlyConfigured(
+                        "Pooling doesn't support persistent connections"
+                    )
+                if pool_options is True:  # simply sets the default options
+                    pool_options = {}
+
+                from psycopg_pool import ConnectionPool
+
+                connect_kwargs = self.get_connection_params()
+                # Ensure we run in autocommit, Django properly sets it later on
+                connect_kwargs["autocommit"] = True
+                pool = ConnectionPool(
+                    kwargs=connect_kwargs,
+                    open=False,  # Do not open the pool during startup
+                    configure=self._configure_connection,
+                    **pool_options,
+                )
+
+            # NOTE: `setdefault` ensures that multiple threads don't set this in
```

Comment:
One subtlety with this trick that I checked is that `pool.open` is thread safe; luckily it is :+1: